tt_metal/graph_tracking.hpp

```diff
@@ -0,0 +1,113 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <string>
+#include <any>
+#include <span>
+#include <string_view>
+
+#include "tt_metal/common/core_coord.h"
+#include "tt_metal/impl/buffers/buffer.hpp"
+
+namespace tt::tt_metal {
+
+    class IGraphProcessor{
+    public:
+        IGraphProcessor() = default;
+
+        virtual void track_allocate(tt::tt_metal::Buffer* buffer, bool bottom_up) {};
+
+        virtual void track_deallocate(tt::tt_metal::Buffer* buffer) {};
+
+        virtual void track_allocate_cb(const CoreRangeSet &core_range_set, uint64_t addr, uint64_t size) {};
+
+        virtual void track_deallocate_cb() {};
+
+        virtual void track_begin_op(std::string_view function_name, std::span<std::any> input_parameters) {};
+
+        virtual void track_end_op(const std::any& output_tensors) {};
+
+
+        virtual void begin_capture(bool fake_run) {};
+        virtual std::string end_capture() {return "";};
```

Comment:
do you think this will return a more stable structure later? 
string (json) for now only to iterate fast on the format?

Reply:
Lets use string now. At least this pass after all your comments.

Reply:
I do t mind at all. Just wonder if you have next steps in mind already